### PR TITLE
feat: Remove inactive group calendars

### PR DESCRIPTION
**Pull request by peterstarling**
Removes inactive calendars from the `/calendars` endpoint.

## Notable changes:
- disables pagination for the calendar endpoint
- migrates REST calendar fetch to GraphQL

see: https://trello.com/c/HhKZss3v/1781-admins-can-mark-groups-as-inactive-and-omit-them-from-calendars-page

### CODE REVIEW

File: apps/asap-server/src/entities/group.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ export const parseGraphQLGroup = (
     return team;
   });
   const calendars: CalendarResponse[] = (item.flatData?.calendars || []).map(
-    (c) => parseGraphQLCalendar(c as Calendars), // @todo remove cast
+    (c) => parseGraphqlCalendar(c as Calendars), // @todo remove cast
```
Comment (kmaid) [inline apps/asap-server/src/entities/group.ts:28]: Can we remove this cast now?

File: apps/asap-server/src/controllers/calendars.ts [hunk 2]
```diff
@@ -30,21 +32,50 @@ export default class Calendars implements CalendarController {
...
 
-  async fetch(options: {
-    take: number;
-    skip: number;
-  }): Promise<ListCalendarResponse> {
-    const { take = 50, skip = 0 } = options;
-    const { total, items: calendars } =
-      await this.calendarSquidexRestClient.fetch({
-        take,
+  async fetch(): Promise<ListCalendarResponse> {
```
Comment (niezgoda) [inline apps/asap-server/src/controllers/calendars.ts:35]: When you have basically refactored this code - why have you not moved the business logic (with specific technologies/API behind) into the proper class or file - like move it away from controller?

File: apps/asap-server/src/controllers/calendars.ts [hunk 3]
```diff
@@ -30,21 +32,50 @@ export default class Calendars implements CalendarController {
...
+      return {
+        total: 0,
+        items: [],
+      };
+    }
+
+    const activeCalendars = calendars.filter(
+      (calendar) =>
+        calendar.referencingGroupsContents &&
+        calendar.referencingGroupsContents.findIndex(
```
Comment (niezgoda) [inline apps/asap-server/src/controllers/calendars.ts:71]: This wouldn't work?
calendar.referencingGroupsContents?.findIndex 
so you could remove "calendar.referencingGroupsContents &&" part

**Review by niezgoda — APPROVED** [review]

File: apps/asap-server/src/controllers/calendars.ts (hunk 2 above)
Comment (peterstarling) [reply apps/asap-server/src/controllers/calendars.ts:35]: good question! I think most of the logic belongs to this class anyway. The controller wraps stuff like declaring queries and getting the data back, the only thing that is extracted (and was) is the parsing logic.

Mind you - we do not have CONTRIBUTING guide that would actually specify which type of logic should belong to controller and what should be moved up, but there is a ticket for it: https://trello.com/c/yCshrAnG/1666-split-be-controllers-into-2-layers

File: apps/asap-server/src/controllers/calendars.ts (hunk 3 above)
Comment (peterstarling) [reply apps/asap-server/src/controllers/calendars.ts:71]: actually it wouldn't - the part before `?` would resolve to undefined which is not equal to `-1` and would therefore pass the condition, but the requirements are that a calendar without a group (ie empty array or no reference to groups at all) should be skipped.

Comment (niezgoda) [reply apps/asap-server/src/controllers/calendars.ts:71]: So you could use sth like calendar.referencingGroupsContents?.findIndex(...) ?? false or so :)

**Pull request merged**